doppler/resource_workplace_role.go

```diff
@@ -0,0 +1,143 @@
+package doppler
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceWorkplaceRole() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceWorkplaceRoleCreate,
+		ReadContext:   resourceWorkplaceRoleRead,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		UpdateContext: resourceWorkplaceRoleUpdate,
+		DeleteContext: resourceWorkplaceRoleDelete,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Description: "The name of the Doppler workplace role",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"permissions": {
+				Description: "A list of [Doppler workplace permissions](https://docs.doppler.com/reference/workplace_roles-create)",
+				Type:        schema.TypeSet,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Required: true,
+			},
+			"identifier": {
+				Description: "The role's unique identifier",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"is_custom_role": {
+				Description: "Whether or not the role is custom (as opposed to Doppler built-in)",
+				Type:        schema.TypeBool,
+				Computed:    true,
+			},
+			"is_inline_role": {
+				Description: "Whether or not the role is inline (as opposed to Doppler built-in)",
```

Comment:
Inline roles are the "virtual" roles that service accounts have if you select custom permissions (rather than a pre-defined custom or built-in role).

It's included in the data transfer object for the API but I think it might make more sense to omit it from these models as it will always be `false` for roles managed here